JoodHajeer/assignment1/Q2

```diff
@@ -0,0 +1,113 @@
+#include <iostream>
+#include <stdio.h>  
+#include <string>
+using namespace std;
+
+template <class T>
+class SLList;
+template <class T>
+class SLLNode{
+//Singly Linked List Node Class building block for a list
+// A single linked list node class to store elements of a generic type T
+public:
+SLLNode(T &VAL, SLLNode* NEXT);
+~SLLNode();
+private:
+	T val;
+	SLLNode * next;
+	friend class SLList<T>;
+};
+template <class T>
+SLLNode<T>::SLLNode(T &VAL, SLLNode * NEXT){
+	val=VAL;
+	next=NEXT;
+}
+template <class T>
+SLLNode<T>::~SLLNode(){}
+
+template <class T>
+class SLList{
+//ADT single linked list class
+//I have implemented only the list functions the algorithm needs.
+public:
+	SLList();
+	~SLList();
+	bool IsEmpty();
+	void addToTail(T&VAL);
+	void clear();
+	int size();
+	T findElementToLast(int k);
+private:
+	SLLNode<T> * head;
+	SLLNode<T> * tail;
+};
+template <class T>
+SLList<T> :: SLList(){
+	head=tail=NULL;
```

Comment:
It's generally preferred to use nullptr instead of NULL, it's typed when NULL is actually 0.